### PR TITLE
Basic processing documentation for CLI.

### PR DESCRIPTION
**Pull request by Garyfallidis**

### CODE REVIEW

File: doc/interfaces/basic_flow.rst [hunk 1]
```diff
@@ -4,23 +4,80 @@
...
+
+Move to the folder with the data::
+
+    $ cd data_folder/cfin_multib
+    $ ls or dir
+
+    dwi.bval
+    dwi.bvec
+    dwi.nii
+    T1.nii
```
Comment (arokem) [inline doc/interfaces/basic_flow.rst]: These are not the file-names here:

```
T1.nii
__DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.bval
__DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.bvec
__DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii
```

File: doc/interfaces/basic_flow.rst [hunk 2]
```diff
@@ -4,23 +4,80 @@
...
+
+Investigate the bvals::
+
+    $ dipy_info dwi.nii
```
Comment (arokem) [inline doc/interfaces/basic_flow.rst]:
```suggestion
    $ dipy_info __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii
```

File: doc/interfaces/basic_flow.rst [hunk 3]
```diff
@@ -4,23 +4,80 @@
...
+
+
+
+Visualize the diffusion data::
+
+    dipy_horizon __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii
```
Comment (arokem) [inline doc/interfaces/basic_flow.rst]: This currently throws an error: 

```
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/bin/dipy_horizon", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/dipy/bin/dipy_horizon", line 10, in <module>
    run_flow(HorizonFlow())
  File "/Users/arokem/source/dipy/dipy/workflows/flow_runner.py", line 90, in run_flow
    return flow.run(**args)
  File "/Users/arokem/source/dipy/dipy/workflows/viz.py", line 59, in run
    for input_output in io_it:
  File "/Users/arokem/source/dipy/dipy/workflows/multi_io.py", line 258, in __iter__
    IO = np.concatenate([ins, out], axis=1)
ValueError: all the input arrays must have same number of dimensions
```

File: doc/interfaces/basic_flow.rst [hunk 4]
```diff
@@ -4,23 +4,80 @@
...
+.. figure:: https://github.com/dipy/dipy_data/blob/master/cfin_basic1.png?raw=true
+    :width: 70 %
+    :alt: alternate text
+    :align: center
+
+    Visualization of a slice from the first volume of the diffusion data
 
 Create brain mask::
 
     dipy_median_otsu dwi.nii.gz --out_dir out_work/
```
Comment (arokem) [inline doc/interfaces/basic_flow.rst]:
```suggestion
    dipy_median_otsu __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii --out_dir out_work/
```

Comment (arokem) [reply doc/interfaces/basic_flow.rst]: Also, the API has changed, so we need to provide a vol_idx: 

```
(dipy) cfin_multib$dipy_median_otsu __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii --out_dir out_work/
INFO:Applying median_otsu segmentation on __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/bin/dipy_median_otsu", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/dipy/bin/dipy_median_otsu", line 9, in <module>
    run_flow(MedianOtsuFlow())
  File "/Users/arokem/source/dipy/dipy/workflows/flow_runner.py", line 90, in run_flow
    return flow.run(**args)
  File "/Users/arokem/source/dipy/dipy/workflows/segment.py", line 73, in run
    autocrop=autocrop, dilate=dilate)
  File "/Users/arokem/source/dipy/dipy/segment/mask.py", line 196, in median_otsu
    raise ValueError("For 4D images, must provide vol_idx input")
ValueError: For 4D images, must provide vol_idx input
```

File: doc/interfaces/basic_flow.rst [hunk 5]
```diff
@@ -4,23 +4,80 @@
...
+    :width: 70 %
+    :alt: alternate text
+    :align: center
+
+    Visualization of a slice from the first volume of brain mask
+
+
+Perform DTI::
 
     dipy_fit_dti dwi.nii.gz dwi.bval dwi.bvec out_work/brain_mask.nii.gz --out_dir out_work/
```
Comment (arokem) [inline doc/interfaces/basic_flow.rst]:
```suggestion
    dipy_fit_dti __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.nii.gz __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.bval __DTI_AX_ep2d_2_5_iso_33d_20141015095334_4.bvec out_work/brain_mask.nii.gz --out_dir out_work/
```

File: doc/interfaces/basic_flow.rst [hunk 6]
```diff
@@ -1,42 +1,241 @@
...
+.. figure:: https://github.com/dipy/dipy_data/blob/master/cfin_basic1.png?raw=true
+    :width: 70 %
+    :alt: alternate text
+    :align: center
+
+    Visualization of a slice from the first volume of the diffusion data
+
+We can use ``dipy_median_otsu`` to build a brain mask for the diffusion data::
+
+    dipy_median_otsu dwi.nii --out_dir out_work/
```
[the remark this answers is not in the document]
Comment (arokem) [reply doc/interfaces/basic_flow.rst]: I am getting 
```
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/bin/dipy_median_otsu", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/dipy/bin/dipy_median_otsu", line 7, in <module>
    run_flow(MedianOtsuFlow())
  File "/Users/arokem/source/dipy/dipy/workflows/flow_runner.py", line 89, in run_flow
    return flow.run(**args)
  File "/Users/arokem/source/dipy/dipy/workflows/segment.py", line 72, in run
    autocrop=autocrop, dilate=dilate)
  File "/Users/arokem/source/dipy/dipy/segment/mask.py", line 195, in median_otsu
    raise ValueError("For 4D images, must provide vol_idx input")
```

[the remark this answers is not in the document]
Comment (arokem) [reply doc/interfaces/basic_flow.rst]: This works when I run: `dipy_median_otsu dwi.nii --out_dir out_work/ --vol_idx 0`

File: doc/interfaces/basic_flow.rst [hunk 7]
```diff
@@ -1,42 +1,241 @@
...
-    dipy_median_otsu dwi.nii.gz --out_dir out_work/
+    dipy_track_det out_work/peaks.pam5 out_work/fa.nii.gz out_work/seed_mask.nii.gz --out_tractogram 'out_work/tracks_from_peaks.trk'
 
-Create stopping criteria::
 
-    dipy_fit_dti dwi.nii.gz dwi.bval dwi.bvec out_work/brain_mask.nii.gz --out_dir out_work/
+We can visualize the result using ``dipy_horizon``. The ``--cluster`` option allows to directly see the clusters of the tractogram::
 
-Create peaks::
+    dipy_horizon out_work/tracts_from_peaks.trk --cluster
```
[the remark this answers is not in the document]
Comment (arokem) [reply doc/interfaces/basic_flow.rst]: I am getting: 

```
(dipy) cfin_multib$dipy_horizon out_work/tracts_from_peaks.trk --cluster
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/bin/dipy_horizon", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/arokem/source/dipy/bin/dipy_horizon", line 7, in <module>
    run_flow(HorizonFlow())
  File "/Users/arokem/source/dipy/dipy/workflows/flow_runner.py", line 89, in run_flow
    return flow.run(**args)
  File "/Users/arokem/source/dipy/dipy/workflows/viz.py", line 104, in run
    for input_output in io_it:
  File "/Users/arokem/source/dipy/dipy/workflows/multi_io.py", line 258, in __iter__
    IO = np.concatenate([ins, out], axis=1)
ValueError: all the input arrays must have same number of dimensions
```

Any ideas?

[the remark this answers is not in the document]
Comment (skoudoro) [reply doc/interfaces/basic_flow.rst]: Sometimes, you have this error when it doesn't find/understand the path... Can you add some double quotes. My last PR improve a bit this point but I think in general we should start to use pathlib module.

[the remark this answers is not in the document]
Comment (arokem) [reply doc/interfaces/basic_flow.rst]: Oh - I get it! It's just a typo. This should be: 

`$ dipy_horizon out_work/tracks_from_peaks.trk --cluster`

File: doc/interfaces/basic_flow.rst [hunk 8]
```diff
@@ -1,42 +1,241 @@
...
 
-    dipy_fit_csd dwi.nii.gz dwi.bval dwi.bvec out_work/brain_mask.nii.gz --out_dir out_work/
+Alternatively, we can create deterministic tracks using the maximum value of a spherical harmonics cone::
 
-Create seeding mask (this creates seed_mask in new folder)::
+    dipy_track_det peaks.pam5 fa.nii.gz seed_mask.nii.gz --out_tractogram 'tracks_from_sh.trk' --use_sh
```
[the remark this answers is not in the document]
Comment (arokem) [reply doc/interfaces/basic_flow.rst]: I believe that the CLI `dipy_track_det` is now `dipy_track_det`. I am also getting: 

`dipy_track_local: error: unrecognized arguments: --use_sh`

**Comment by arokem** [comment]
You should add this file in the index of this section: 

https://github.com/dipy/dipy/blob/master/doc/interfaces/index.rst

And make sense that this and the already-existing example make sense.

**Comment by skoudoro** [comment]
Hi @Garyfallidis,

What is the status of this PR? Can you address @arokem comments? Is it close to be done?

**Comment by Garyfallidis** [comment]
Hello I have addressed the comments and fixed some other things too.

I think this is okay for now as a basic introduction to our workflows but we should soon used a better (whole brain and without strange filenames) dataset.  For this reason I also opened and issue to not forget. Here it is https://github.com/dipy/dipy/issues/2138

File: doc/interfaces/basic_flow.rst [hunk 9]
```diff
@@ -1,42 +1,248 @@
...
+
+.. figure:: https://github.com/dipy/dipy_data/blob/master/cfin_basic2.png?raw=true
+    :width: 70 %
+    :alt: alternate text
+    :align: center
+
+    Visualization of a slice from the generated brain mask
+
+
+Perform DTI using ``dipy_fit_dti`. The input of this function is the DWI data, b-values and b-vector files and the brain mask that we calculated in the previous step::
```
Comment (skoudoro) [inline doc/interfaces/basic_flow.rst:177]: typo: 1 back stick to remove on the left of `dipy_fit_dti `

File: doc/interfaces/basic_flow.rst [hunk 10]
```diff
@@ -1,42 +1,248 @@
...
+    :align: center
 
-    dipy_fit_dti dwi.nii.gz dwi.bval dwi.bvec out_work/brain_mask.nii.gz --out_dir out_work/
+    Showing tracks from the specific dataset. This dataset contains only a few slices.
 
-Create peaks::
+We can visualize the result using ``dipy_horizon``. The ``--cluster`` option allows to directly see the clusters of the tractogram::
 
-    dipy_fit_csd dwi.nii.gz dwi.bval dwi.bvec out_work/brain_mask.nii.gz --out_dir out_work/
+    dipy_horizon out_work/tracts_from_peaks.trk --cluster
```
Comment (skoudoro) [inline doc/interfaces/basic_flow.rst:216]: typo: `tracks_from_peaks.trk` instead of `tracts_from_peaks.trk`

File: doc/interfaces/basic_flow.rst [hunk 11]
```diff
@@ -1,42 +1,248 @@
...
 
-Create seeding mask (this creates seed_mask in new folder)::
+Alternatively, we can create deterministic tracks using the maximum value of a spherical harmonics cone::
 
-    dipy_mask out_work/fa.nii.gz 0.4 --out_dir out_work/ --out_mask seed_mask.nii.gz
+    dipy_track_local peaks.pam5 fa.nii.gz seed_mask.nii.gz --out_tractogram 'tracks_from_sh.trk' --use_sh
```
Comment (skoudoro) [inline doc/interfaces/basic_flow.rst:220]: `--use_sh` parameter does not exist anymore. You should use `--tracking_method` and choose your method between:
- "eudx" 
- "deterministic" or "det" for a deterministic tracking
- "probabilistic" or "prob" for a Probabilistic tracking
- "closestpeaks" or "cp" for a ClosestPeaks tracking

Furthermore, `peaks.pam5` is not on the current folder, so you need to do `out_work/peaks.pam5`. The correct command line is:
`dipy_track_local out_work/peaks.pam5 out_work/fa.nii.gz out_work/seed_mask.nii.gz --out_tractogram 'tracks_from_sh.trk' --tracking_method eudx`

File: doc/interfaces/basic_flow.rst [hunk 12]
```diff
@@ -1,42 +1,248 @@
...
+----------
 
-Create tracks using sh cone::
+.. [Garyfallidis14] Garyfallidis, E., M. Brett, B. Amirbekian, A. Rokem,
+    S. Van Der Walt, M. Descoteaux, and I. Nimmo-Smith.
+    "DIPY, a library for the analysis of diffusion MRI data".
+    Frontiers in Neuroinformatics, 1-18, 2014.
 
-    dipy_track_det peaks.pam5 fa.nii.gz seed_mask.nii.gz --out_tractogram 'tracks_from_sh.trk' --use_sh
+.. [Aganj2010] Aganj I, Lenglet C, Sapiro G, Yacoub E, Ugurbil K, Harel N.
```
Comment (skoudoro) [inline doc/interfaces/basic_flow.rst:238]: this citation is never referenced

File: doc/interfaces/basic_flow.rst [hunk 13]
```diff
@@ -1,42 +1,248 @@
...
+   "Reconstruction of the orientation distribution function in single- and
+   multiple-shell q-ball imaging within constant solid angle", Magnetic
+   Resonance in Medicine. 2010 Aug;64(2):554-66. doi: 10.1002/mrm.22365
 
+.. [Garyfallidis12] Garyfallidis E., "Towards an accurate brain tractography",
```
Comment (skoudoro) [inline doc/interfaces/basic_flow.rst:243]: this citation is never referenced

**Comment by Garyfallidis** [comment]
Thank you for your review. I believe it is corrected now. Let me know otherwise.

**Comment by skoudoro** [comment]
Hi @Garyfallidis,

There are still 2 comments not addressed. The command line is failing because of a typo. Can you see above my comment?
Thank you

**Comment by skoudoro** [comment]
Thank you @Garyfallidis! merging

**Pull request merged**